--- guile-gnome-platform-2.15.92/corba/gnome/corba/Makefile.am.orig	2007-05-20 18:39:16.000000000 +0200
+++ guile-gnome-platform-2.15.92/corba/gnome/corba/Makefile.am	2007-05-26 22:07:43.189771490 +0200
@@ -15,7 +15,7 @@
 libguile_gnome_corba_@API_VERSION@_la_CFLAGS = $(CORBA_CFLAGS)  \
   $(AM_CFLAGS) $(GUILE_GLIB_CFLAGS) $(GUILE_CFLAGS) -DORBIT2_STUBS_API
 
-libguile_gnome_corba_@API_VERSION@_la_LIBADD = $(GUILE_GLIB_LIBS) $(GUILE_LIBS)
+libguile_gnome_corba_@API_VERSION@_la_LIBADD = $(CORBA_LIBS) $(GUILE_GLIB_LIBS) $(GUILE_LIBS)
 libguile_gnome_corba_@API_VERSION@_la_LDFLAGS = -export-dynamic # FIXME: version info
 
 guile_gnome_includedir = $(includedir)/guile-gnome-@API_VERSION@
--- guile-gnome-platform-2.16.4/gnome-vfs/gnome/gw/Makefile.am.orig	2015-01-24 15:07:42.000000000 +0100
+++ guile-gnome-platform-2.16.4/gnome-vfs/gnome/gw/Makefile.am	2016-02-29 18:57:24.314595083 +0100
@@ -20,7 +20,7 @@
 	$(AM_CFLAGS) $(GNOME_VFS_CFLAGS) $(GUILE_CFLAGS) \
 	$(G_WRAP_CFLAGS) $(GNOME_VFS_DEPRECATED_CFLAGS)
 
-libgw_guile_gnome_gnome_vfs_la_LIBADD = $(GNOME_VFS_LIBS) $(GUILE_LIBS) \
+libgw_guile_gnome_gnome_vfs_la_LIBADD = $(GNOME_VFS_LIBS) -lbonobo-activation $(GUILE_LIBS) \
 	$(G_WRAP_LIBS) $(GUILE_GLIB_LIBS)
 
 libgw_guile_gnome_gnome_vfs_la_LDFLAGS = -module
